experiment/manual-trigger/manual-trigger.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/url"
+	"strings"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/jenkins"
+	"k8s.io/test-infra/prow/kube"
+)
+
+type options struct {
+	githubEndpoint     string
+	githubToken        string
+	jenkinsBearerToken string
+	jenkinsEndpoint    string
+	jobName            string
+	num                int
+	org                string
+	repo               string
+}
+
+func flagOptions() options {
+	o := options{}
+	flag.StringVar(&o.githubEndpoint, "githubapiurl", "https://api.github.com", "GitHub's API endpoint.")
+	flag.StringVar(&o.jenkinsEndpoint, "jenkinsurl", "", "Jenkins url.")
+
+	flag.StringVar(&o.githubToken, "ghtoken", "", "Path to file containing GitHub token.")
+	flag.StringVar(&o.jenkinsBearerToken, "jbtoken", "", "Path to file containing Jenkins Bearer token.")
+
+	flag.StringVar(&o.jobName, "jobname", "", "Name of Jenkins job")
+
+	flag.IntVar(&o.num, "num", 0, "GitHub issue number")
+	flag.StringVar(&o.org, "org", "", "GitHub organization")
+	flag.StringVar(&o.repo, "repo", "", "GitHub repository")
+	flag.Parse()
+	return o
+}
+
+func sanityCheckFlags(o options) error {
+	if o.num <= 0 {
+		return fmt.Errorf("empty or invalid --num")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("empty --org")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("empty --repo")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("empty --ghtoken")
+	}
+	if o.jenkinsBearerToken == "" {
+		return fmt.Errorf("empty --jbtoken")
+	}
+	if o.jenkinsBearerToken == "" {
+		return fmt.Errorf("empty --jobname")
+	}
+
+	_, err := url.Parse(o.githubEndpoint)
+	if o.githubEndpoint == "" || err != nil {
+		return fmt.Errorf("must specify a valid URL to --githubApiUrl")
+	}
+
+	_, err = url.Parse(o.jenkinsEndpoint)
+	if o.jenkinsEndpoint == "" || err != nil {
+		return fmt.Errorf("must specify a valid URL to --jenkinsurl")
+	}
+
+	return nil
+}
+
+func readToken(path string) (string, error) {
+	contents, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(contents)), nil
+}
+
+func main() {
+	o := flagOptions()
+	err := sanityCheckFlags(o)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	jbtoken, err := readToken(o.jenkinsBearerToken)
+	if err != nil {
+		log.Fatalf("cannot read file specified by --jbtoken %v", err)
+	}
+
+	ghtoken, err := readToken(o.githubToken)
+	if err != nil {
+		log.Fatalf("cannot read file specified by --ghtoken: %v", err)
+	}
+
+	auth := jenkins.BearerTokenAuthConfig{
+		Token: jbtoken,
+	}
+
+	authconfig := jenkins.AuthConfig{
+		Basic:       nil,
+		BearerToken: &auth,
+	}
+
+	jc := jenkins.NewClient(o.jenkinsEndpoint, &authconfig)
+
+	gc := github.NewClient(ghtoken, o.githubEndpoint)
+
+	pr, err := gc.GetPullRequest(o.org, o.repo, o.num)
+	if err != nil {
+		log.Fatalf("Unable to get information on pull request (org=%q repo=%q pr-num=%d): %v", o.org, o.repo, o.num, err)
```

Comment:
We usually use `Printf("%s/%s#%d", org, repo, num)` to format this triplet of info

Reply:
The use of Fatalf seems fine here.

Reply:
Errr I just meant the way the three pieces of information were formatted